fronts-client/src/services/pandaFetch.ts

```diff
@@ -16,26 +19,45 @@ const pandaFetch = (
   count: number = 0
 ): Promise<Response> =>
   new Promise(
-    async (resolve: (r: Response) => any, reject: (r: Response) => any) => {
-      const res = await fetch(url, {
-        ...options,
-        credentials: 'same-origin',
-      });
+    async (resolve: (r: Response) => any, reject: (e: any) => any) => {
+      try {
+        const res = await fetch(url, {
+          ...options,
+          credentials: 'same-origin',
+        });
 
-      if ((res.status === 419 || res.status === 401) && count < 1) {
-        try {
-          await reEstablishSession(reauthUrl, 5000);
-          const res2 = await pandaFetch(url, options, count + 1);
-          return resolve(res2);
-        } catch (e) {
-          notifications.notify({ message: reauthErrorMessage, level: 'error' });
-          return reject(e);
+        if ((res.status === 419 || res.status === 401) && count < 1) {
+          try {
+            await reEstablishSession(reauthUrl, 5000);
+            const res2 = await pandaFetch(url, options, count + 1);
+            return resolve(res2);
+          } catch (e) {
+            notifications.notify({
+              message: reauthErrorMessage,
+              level: 'error',
+            });
+            return reject(
+              isError(e) ? `Auth Issue (${e ? e.toString() : ''})` : e
+            );
+          }
+        } else if (res.status < 200 || res.status >= 300) {
+          notifications.notify({
+            message:
+              'Request failed. Your changes may not be saved. Please wait or reload the page.',
+            level: 'error',
+          });
+          return reject(res);
         }
-      } else if (res.status < 200 || res.status >= 300) {
-        return reject(res);
-      }
 
-      return resolve(res);
+        return resolve(res);
+      } catch (error) {
+        notifications.notify({
```

Comment:
This change is a Good Thing, but we've no way at present to assess its impact on users.

I'd recommend capturing this failure in Raven (it's got retry strats, this isn't as paradoxical as it sounds 😁 ) so we can see if users are experiencing high volumes of these messages, and diagnose as appropriate.

There's an example in `move.ts`, e.g. `Raven.captureMessage(`Moving item up in clipboard failed: ${e.message}`);` (but let's not chuck away the stack trace!)

In any case, great catch. Pun not intended.

Reply:
happy to add the call to Sentry, are you recommending that instead of a user banner?

Reply:
added in addition to the banners (as discussed over chat) - see https://github.com/guardian/facia-tool/pull/1277/commits/5884f19903acdf639dff6849012a242567f8e32d